contracts/Sweep/TransferApprover/VestingApprover.sol

```diff
@@ -153,28 +114,22 @@ contract VestingApprover is ITransferApprover, Ownable {
         address from,
         address to,
         uint256 amount
-    ) external onlySweepr returns (bool) {
-        // Check whitelist if approver state is 'Closed'
+    ) external view onlySweepr returns (bool) {
+        // allow minting & burning & tansfers from sender not in vesting list
         if (
-            isClosed &&
-            (from == address(0) || to == address(0) || whitelists[to])
+            from == address(0) ||
+            to == address(0) ||
+            from != vestingSchedules[from].beneficiary
         ) return true;
 
         // Check if sender has enough balancer
-        if (sweepr.balanceOf(from) < amount) return false;
-        // Check if sender is not be beneficiary
-        if (vestingSchedules[from].beneficiary != address(0)) return false;
-        // Check if receiver is valid beneficiary
-        if (vestingSchedules[to].beneficiary == address(0)) return false;
-
-        VestingSchedule storage vestingSchedule = vestingSchedules[to];
-        uint256 vestedAmount = _computeTransferableAmount(vestingSchedule);
+        uint256 senderBalance = sweepr.balanceOf(from);
+        if (senderBalance < amount) return false;
 
-        if (vestedAmount < amount) return false;
+        VestingSchedule storage vestingSchedule = vestingSchedules[from];
```

Comment:
unnecessary use of storage, please use memory.

Reply:
I removed storage and replace it to memory